crates/bevy_ecs/src/system/query.rs

```diff
@@ -275,6 +275,40 @@ where
         }
     }
 
+    /// Creates a new [`Query`] from a [`QueryState`]
+    ///
+    /// [`Query`] is a simply a borrow of [`QueryState`],
+    /// but when working directly with the [`World`]
+    /// we need to store the query's internal data somewhere, and so cannot return a [`Query`] directly.
```

Comment:
This sentence doesn't quite make sense.

I assume we're trying to say something closer to `Query = QueryState + World`?

This is implying that `Query = &mut QueryState`, which doesn't explain why you actually need `World`

Reply:
> I assume we're trying to say something closer to Query = QueryState + World?

Yeah; that's a better explanation!